### ISSUE TITLE
elipse modifying object does not set the ellipse to it's original value after expiring

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#3172]
A unit with the object below should have another ellipse for on scenario.
Using inspect one can see that units usually have no ellipse attribute. Inspect shows that the unit has the object, which expires correctly, and gets additionally an `ellipse` attribute which remains. 

A possible fix may be to clear the ellipse attribute when the object expires.

```
            [object]
                duration=scenario
                [effect]
                    apply_to=ellipse
                    ellipse=ellipse/unprivileged
                [/effect]
            [/object]
```

**Comment by gfgtdf** [comment]
Looks easy to fix, but the question  is whether changing this would break add-ons that relied on the old behavior

**Comment by sevu** [comment]
The possibility to remove objects is new, at least that won't break it.

No other breaking options come to mind...

**Comment by gfgtdf** [comment]
No, my point was that the same code that would 'reset' those changes would the of course also apply to changes that were made via unstore unit. and add-ons that relied on those changes being persistent might no longer work as expected

**Comment by sevu** [comment]
I'm not sure I exactly understand _where_ the conflict could be - some facts first:

Testing showns that clearing the variable while the object is active results in no change, so UMC couldn't have done that. With code as
_storing the unit_
{CLEAR VARIABLE var.ellipse}
_unstoring the unit_

My thought would be that if an object expires _and_ it has an effect with _aply_to ellipse_, then the same as above is done additionally. In case other `objects` with _apply_to_ it will result in them still working, and UMC cannot otherwise change the ellipse key in the time being either.

What will not work is to set the value to the ellipse to what has been active before the object was, as the object did overwrite that, but at least that's not a regression.

(setting the ellipse to empty ("") instead has the same behaviour as never modifying it, but leaves an empty ellipse attribute)

If you want to try out sth., NR S5b might be a good place.

**Comment by CelticMinstrel** [comment]
What exactly does unstoring a unit have to do with this? Is unstoring a rebuild point? I didn't think it was.

**Comment by jostephd** [comment]
Reproduced in master with:

```
diff --git a/data/scenario-test.cfg b/data/scenario-test.cfg
index 784f11193a6..2bffe40fd5d 100644
--- a/data/scenario-test.cfg
+++ b/data/scenario-test.cfg
@@ -2927,25 +2927,16 @@ For game purposes, the races group into factions; for example, orcs often cooper
         first_time_only=no
         [filter]
             side=1
-            x=12
-            y=6
-            type="Thief"
         [/filter]
-        [object]
-            id=bleach
-            name= "Potion of bleach"
-            image=items/potion-red.png
-            duration=scenario
-            description="Oooh tingly!!!"
-            [effect]
-                apply_to=image_mod
-                add="RC(thief>blonde)"
 
-                [color_palette]
-                    thief=85A7CA,496697,D7f0F0,B9CEE4,293D5F,617CAB
-                [/color_palette]
-            [/effect]
-        [/object]
+            [object]
+                #duration=scenario
+                duration=turn end
+                [effect]
+                    apply_to=ellipse
+                    ellipse=ellipse/unprivileged
+                [/effect]
+            [/object]
     [/event]
 
     [event]
```

**Comment by gfgtdf** [comment]
yes I noticed the same when working on  #4088 . I'm considering making this https://github.com/wesnoth/wesnoth/blob/1.14.7/src/units/unit.cpp#L911 unconditional.
